File: BackEnd/TicketBooking.Domain/Entities/Payment.cs
using TicketBooking.Domain.Common;
using TicketBooking.Domain.Enums;

namespace TicketBooking.Domain.Entities;

public class Payment : AuditableEntity
{
    public Guid OrderId { get; set; }
    public Order Order { get; set; } = null!;

    public decimal Amount { get; set; }
    public PaymentMethod PaymentMethod { get; set; }
    public PaymentStatus Status { get; set; } = PaymentStatus.PENDING;
    public string? TransactionId { get; set; }
}

File: frontend/ticket-booking-web/src/domain/entities/Payment.ts
import type { PaymentMethod } from '../enums/PaymentMethod';
import type { PaymentStatus } from '../enums/PaymentStatus';

export interface Payment {
  id: string;
  orderId: string;
  amount: number;
  paymentMethod: PaymentMethod;
  status: PaymentStatus;
  transactionId?: string;
  createdAt: string;
  updatedAt?: string;
}
